src/lib.rs

```diff
@@ -70,35 +70,65 @@
 //!
 //! ```{rust}
 //! use semver::Version;
+//! use std::error::Error;
 //!
-//! let mut bugfix_release = Version::parse("1.0.0").unwrap();
-//! bugfix_release.increment_patch();
+//! fn try_increment_patch() -> Result<(), Box<Error>> {
```

Comment:
I think we should hide the `main` function and signature of `try_increment_patch` so there's less noise in the examples. You can do this by adding a `#` at the start of a doc test line. Like:

```
//! # try_main() -> Result<(), Box<Error>> {
//! let mut bugfix_release = Version::parse("1.0.0")?;
//! ...
//! # Ok(())
//! # }
//! # fn main() { try_main().unwrap(); }
```

That way you only see the body of the `try_main` function, but the `?` magically works.